clang-tools-extra/clang-tidy/daedalean/IncludeOrderCheck.cpp

```diff
@@ -161,11 +161,15 @@ void IncludeOrderPPCallbacks::InclusionDirective(
     LLVM_DEBUG(llvm::dbgs() << "Invalid extension\n");
   }
 
+  const auto FileEntry = SM.getFileEntryForID(SM.getFileID(HashLoc));
+
+  if (!FileEntry) {
+    // This is not regular file
+    return;
+  }
+
   const std::string CurrentDir =
-      SM.getFileManager()
-          .getCanonicalName(
-              SM.getFileEntryForID(SM.getFileID(HashLoc))->getDir())
-          .str();
+      SM.getFileManager().getCanonicalName(FileEntry->getDir()).str();
```

Comment:
This seems unrelated, could you explain what the problem is here?

Reply:
It crashes on preprocessed source from clion because there are few fake includes that has no files and passed via other methods